src/services/apimap-field-builder.js

```diff
@@ -6,7 +6,13 @@ function ApimapFieldBuilder(model, column, options) {
 
   function isRequired() {
     // eslint-disable-next-line
-    return column._autoGenerated !== true && column.allowNull === false;
+    return column._autoGenerated !== true 
+      && (column.allowNull === false || column.primaryKey);
+  }
+
+  function isGenerated(hasFunctionDefaultValue) {
+    // eslint-disable-next-line no-underscore-dangle
+    return Boolean(column._autoGenerated || hasFunctionDefaultValue);
```

Comment:
Following my comment here:
https://github.com/ForestAdmin/forestadmin-server/pull/1373#discussion_r480963164

I am wondering if:
```suggestion
    return Boolean(column._autoGenerated);
```
shouldn't be enough to track generated ids and technical dates?

Reply:
@arnaudbesnier [my comment](https://github.com/ForestAdmin/forestadmin-server/pull/1373#discussion_r480983186)

In addition to my comment, I can think of an UUID column for instance, that cannot be generated by mysql on the fly when declaring a table. A simple solution could be to use a function to generate a UUID when creating an entry.

In this case, we need to detect that a function is used to generate the value.